security_monkey/datastore.py

```diff
@@ -62,16 +70,37 @@ class Account(db.Model):
     id = Column(Integer, primary_key=True)
     active = Column(Boolean())
     third_party = Column(Boolean())
-    name = Column(String(32))
+    name = Column(String(32), unique=True)
+    s3_name = Column(String(64)) # (deprecated-custom)
+    number = Column(String(12))  # (deprecated-identifier) Not stored as INT because of potential leading-zeros.
     notes = Column(String(256))
-    s3_name = Column(String(64))
-    number = Column(String(12))  # Not stored as INT because of potential leading-zeros.
+    identifier = Column(String(256))  # Unique id of the account, the number for AWS.
     items = relationship("Item", backref="account", cascade="all, delete, delete-orphan")
     issue_categories = relationship("AuditorSettings", backref="account")
-    role_name = Column(String(256))
+    role_name = Column(String(256)) # (deprecated-custom)
```

Comment:
These fields are marked as deprecated, but still referenced throughout the codebase.
`s3_name`, `number`, `role_name`.

I'm worried about storing data in multiple places because inconsistencies could occur. 

I see the migration is copying `account.number` to the `account.identifier`, but it is not copying `account.s3_name` or `account.role_name` to a custom field.

I think we should delete these fields instead of deprecating them, and use the upgrade script to move the existing values into the custom fields. (And the reverse in the downgrade script.)

All the tests and the rest of the project will need to be updated to use the new identifier and custom fields.  Maybe you could override the `Account.__getattr__()` to check the custom fields.

Thoughts?